### DISCUSSION TITLE
default_authentication_plugin=mysql_native_password always automatically added to my.ini

### DISCUSSION THREAD

**Discussion by ppfoong** [leokhoa/laragon#804] [Q&A]
Recently I tried to upgrade MySQL to 8.4.0 but it won't run.

I noticed that this line
default_authentication_plugin=mysql_native_password

is always added to my.ini whenever MySQL server is started, even if I have deleted it or commented it out in my.ini

As MySQL 8.4.0 no longer support this syntax, it won't be able to run. The new syntax for MySQL 8.4.0 is
mysql-native-password=ON

Also, this line will force MySQL 8 to use mysql_native_password instead of the recommended caching_sha2_password

How to disable the auto-writing of default_authentication_plugin=mysql_native_password into my.ini?

**Comment by improved-software** [comment]
## Make `my.ini` **Read-only**

### Prevents it being overwritten

- Make your changes
- Save
- Edit `my.ini` properties  - *Set to **Read-only***

---

![image](https://github.com/user-attachments/assets/41d440e0-8b1f-4035-868e-cf1c8a9debc4)

**Comment by FlavioSuarez** [comment]
You may also have to set the path to your data, something like:
datadir= "C:/laragon6/data/mysql-8.4.0"
